### DISCUSSION TITLE
Much needed feature Custom preset sequencing actions

### DISCUSSION THREAD

**Discussion by dalin-sourcecode** [lllyasviel/Fooocus#3449] [General]
Dear Mashbit, 

I love fooocus, everything that it has become and will be. I am writing to you as a cry for help in the hopes that this feature could come to life. The improvements on enhancing an image have been monumental to say the least. However there is one feature that is badly needed. Custom preset sequencing. Being able to save my presets, all of the advanced settings and being able to run them in sequence. For example, in my workflow I start off by generating an image, that image is then moved to the image prompt or vary then a total new series of settings samplers, scheduler, styles, guidance etc... is then applied to generate a new look. This seems to be possible in COMFYUI but I'm in too deep with FOOOCUS to turn my back now. Is there a way to save presets >>>> run them in sequence >>>> designate the action in between whether it be image prompt, variation, upscale. 

1. GENERATE IMAGE 1>>> with preset 1         
2. IMAGE 1 MOVED TO IMAGE PROMPT OR VARIATION/UPSCALE >>>> with preset 2 
3. GENERATE IMAGE 2 now with IMAGE 1 AS IMAGE PROMPT OR VARIATION/UPSCALE  >>> with preset 2

Thanks for all of the work that you do. I know that your schedule is busy and this is probably a half-baked idea so I wouldn't lose sleep over this. I have it pretty worked out for now. I have a routine of of just pasting in the parameters from the history and it works pretty well.

**Comment by mashb1t** [comment]
> Is there a way to save preset

@zeuseclipse There currently isn't, at least not without manual steps.
Also not all parameters are supported in the preset, which does not allow e.g. any images / masks to be saved/loaded in configs.
Some ideas were already mentioned using a queue system and some forks tried to implement this, but without major adjustments to Gradio queueing and/or introduction of a database or some kind of separate queue managemebt this doesn't work.

=> as i'm doing this in my spare time i have currently no time to implement this feature. Feel free to shoot your shot and implement it, then open a PR to provide the code to thr community.
